modules/core/keeper/msg_server.go

```diff
@@ -726,22 +726,92 @@ func (k Keeper) ChannelUpgradeInit(goCtx context.Context, msg *channeltypes.MsgC
 		return nil, errorsmod.Wrapf(err, "channel upgrade init callback failed for port ID: %s, channel ID: %s", msg.PortId, msg.ChannelId)
 	}
 
-	msg.ProposedUpgradeChannel.Version = version
```

Comment:
Why was this removed? In `WriteUpgradeInitChannel` the version of the channel [is updated with the one in the proposed upgrade channel](https://github.com/cosmos/ibc-go/blob/cian/issue%231619-implement-channelupgradetry-functionality-for-04-channel/modules/core/04-channel/keeper/upgrade.go#L97), but without this, we wouldn't take into account the version returned by the application callback. Maybe instead of modifying the proposed upgrade channel, the version can be passed as a separate argument, as it is done in this PR for `WriteUpgradeTryChannel`.

Reply:
It looks like this is an oversight from a refactor that we did to [`WriteUpgradeTryChannel`](https://github.com/cosmos/ibc-go/blob/00a313677d15572d3677802088ee885e4ed410c6/modules/core/keeper/msg_server.go#L807)

that refactor was intended to be more explicit by not modifying the input message and passing the version directly. I think we should still not modify the message, but we should pass the version into WriteUpgradeInit in the same way we do in the linked code.